### DISCUSSION TITLE
How to cache intermediate results for use by multiple feature calculators?

### DISCUSSION THREAD

**Discussion by atpage** [blue-yonder/tsfresh#884] [Q&A]
I have a handful of custom features that all rely on the same intermediate results (for a given series, and set of parameters).  Rather than recompute these same intermediates in each individual feature calculator, I'd like to compute them once, and allow the other calculators to pick up from the intermediate results (if available).

At first glance, I thought that the `combiner` type was what I was after, but that appears to be for computing results of multiple parameters for a single feature.  I want to do the opposite, compute multiple features from a single set of parameters.

Is there a suggested way to accomplish this?  Thanks!

**Comment by nils-braun** [comment]
Hi @atpage!
That is a very nice question and I think you did already do some good research. The `combiner` type is indeed the correct one to use. For example, have a look into the [`matrix_profile`](https://github.com/blue-yonder/tsfresh/blob/main/tsfresh/feature_extraction/feature_calculators.py#L2369) feature. It first calculates the rather heavy matrix profiles themselves (that would be your intermediate results) and then derives a handful of simple features from it (mean, min, max, etc.).
The trick is, that the parameters for the calculated feature contain the normal set of parameters (in this case: which are needed to calculate the heavy matrix profiles) and an additional parameter called "feature" which indicates which of the features to calculate (min, max, mean, etc.) 
The [`fft_aggregated`](https://github.com/blue-yonder/tsfresh/blob/main/tsfresh/feature_extraction/feature_calculators.py#L1118) is quite similar.

**Reply by atpage** [reply]
Ah, I see, so `feature` would actually be a parameter, and the calculator function would receive all the parameters at once.  I can work with that.  Thanks again!
